### PR TITLE
Add LLM-based US documentation page

### LINKED ISSUE #175: Include LLM-based US in documentation

### PR DESCRIPTION
**Pull request by NoB0**
Fixes #175

### CODE REVIEW

File: docs/source/llm_based.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+The LLM interface is responsible for interacting with the large language model to generate responses. Currently, two LLM interfaces are supported: Ollama and OpenAI. 
+
+Ollama
+^^^^^^
+
+:py:class:`usersimcrs.simulator.llm.interfaces.ollama_interface.OllamaLLMInterface`
+
+This interface is used to interact with a LLM that is hosted on the `Ollama platform <https://ollama.com>`_. The interface sends requests to the `Ollama API <https://github.com/ollama/ollama/blob/main/docs/api.md>`_ to generate the responses. 
+
+This interface is configured with a YAML file that includes: the model name, the host URL, whether to stream the responses, and the LLM specific options. An example configuration is shown below:
```
Comment (kbalog) [inline docs/source/llm_based.rst:33]: Should we mention where these YAML config files are located?

**Review by kbalog — APPROVED** [review]
LGTM

File: docs/source/llm_based.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
+LLM-based simulator
+===================
+
+This simulator relies on a large language model (LLM) to generate utterances. Currently, a single zero-shot prompt is supported for the response generation. The interactions with the LLM are managed by the LLM interface.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/llm_based.rst:4]: Done.

File: docs/source/llm_based.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+Prompt
+------
+
+:py:class:`usersimcrs.simulator.llm.prompt.Prompt`
+
+The prompt is inspired by the work of Terragni et al. [1]_ It includes the following information: task description, information need, optionally the simulated user persona, and conversational context. The prompt is built as follows:
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/llm_based.rst:11]: I like the reference suggestion, I modified the other documentation page accordingly.
I rephrase the sentence to remove ambiguity on what is optional in the prompt.

**Pull request merged**